components/scream/src/physics/p3/tests/p3_saturation_unit_tests.cpp

```diff
@@ -114,44 +188,210 @@ struct UnitWrap::UnitTest<D>::TestP3Saturation
     /*Originally written by Kyle Pressel, updated by Peter Caldwell on 4/5/20.
      *This code tests polysvp1 and qv_sat at 0 degrees C, at a very cold T, and at a very hot T
      *to make sure our impl gets the same answer as Flatau et al 1992:
-     *(https://journals.ametsoc.org/doi/abs/10.1175/1520-0450%281992%29031%3C1507%3APFTSVP%3E2.0.CO%3B2
+     *(https://journals.ametsoc.org/jamc/article/31/12/1507/14870/Polynomial-Fits-to-Saturation-Vapor-Pressure)
      *For 0 degrees C, polysvp values can be read directly from Flatau. For other cases, I independently
-     *coded up the Flatau scheme in python and used it to derive the expected values. My python code is
+     *coded up the Flatau scheme (polysvp1) in python and used it to derive the expected values. My python code is
      *in https://github.com/E3SM-Project/scream-docs.git analysis-scripts/test_qv_sat.py
      */
 
-
     int nerr = 0;
     TeamPolicy policy(util::ExeSpaceUtils<ExeSpace>::get_default_team_policy(1, 1));
     Kokkos::parallel_reduce("TestTableIce::run", policy, KOKKOS_LAMBDA(const MemberType& team, int& errors) {
 
       errors = 0;
-      const auto tmelt = C::Tmelt;
 
-      // Just Freezing Case: Test values @ 273.15K @ 1e5 Pa
+      //Following struct stores data for lauching the saturation_tests function
+      //for polysvp1 and MurphyKoop(MK) saturation schemes. We are supplying temperature,
+      //pressure, expected values in double and single precision. Polysvp1 is very
+      //sensitive (due to shear number of calcs required at low temps)to single
+      //precision values at low temperatures, therefore we explicity supply single
+      //precision expected values. Single precision expected values are computed
+      //from a python code using numpy's float32. The code uses the same functions
+      //for polysvp1 and MK as in the code present at:
+      // https://github.com/E3SM-Project/scream-docs.git analysis-scripts/test_qv_sat.py
+
+      //total number expected vals to store for each precision for both MK and polysvp1
+      const int ncrv = 8; // 4 for each saturation scheme
+
+      struct sat_test_args {
```

Comment:
Great job putting the info in a struct and handling single and double precision separately!